Frontend release 0.0.22

hannalaakso opened the pull request:
Update with all the latest changes from source:

Breaking changes:

- The link styles in the core layer no longer style `a` elements directly, but
  instead provide a `govuk-link` class which you will need to apply to links
  individually. (PR [#427](https://github.com/alphagov/govuk-frontend/pull/427))
- The link component has been removed from Frontend as the link styles have
  been moved to the core
  (PR [#431](https://github.com/alphagov/govuk-frontend/pull/431))
- Rename `legal-text` argument accepted by `legal-text` component to `text` (PR [#431](https://github.com/alphagov/govuk-frontend/pull/432))
- Rename `legal-text` component to `warning-text` (PR [#431](https://github.com/alphagov/govuk-frontend/pull/432))

New features:

- The prose scope has been extended to style links, which means links within the
  scope do not need the `govuk-link` class applied.
  (PR [#427](https://github.com/alphagov/govuk-frontend/pull/427))
- The muted link variant from the link component is now available as a core
  class (`govuk-link--muted`).
  (PR [#427](https://github.com/alphagov/govuk-frontend/pull/427))

Fixes:

- The error summary component allows users to pass HTML for an entry in the list
  again. (PR [#428](https://github.com/alphagov/govuk-frontend/pull/428))
- Error list entries in the error summary component no longer get wrapped in
  links when no `href` is provided.
  (PR [#428](https://github.com/alphagov/govuk-frontend/pull/428))
- Remove redundant 'resets' for link print styles
  (PR [#427](https://github.com/alphagov/govuk-frontend/pull/427))
- The back link, breadcrumbs, error summary, previous / next and skip link
  components have been updated to include explicit link styling, as they
  previously relied on the global link styles.
  (PR [#427](https://github.com/alphagov/govuk-frontend/pull/427))
- Adjust `warning-text` icon by 1px for New Transport

Internal:

- Add prose scope example (PR [#429](https://github.com/alphagov/govuk-frontend/pull/429))
- Links within the review app and the examples have been updated to use the
  `govuk-link` class.
  (PR [#427](https://github.com/alphagov/govuk-frontend/pull/427))
- Improve documentation around publishing (PR [#430](https://github.com/alphagov/govuk-frontend/pull/430))  
- Improve documentation around contributing (PR [#433](https://github.com/alphagov/govuk-frontend/pull/433))

36degrees commented on packages/all/node_modules/@govuk-frontend/back-link:
```diff
@@ -0,0 +1 @@
+../../../back-link
```
Where did these files come from? Are these meant to be here?

36degrees commented on packages/back-link/node_modules/@govuk-frontend/globals:
```diff
@@ -0,0 +1 @@
+../../../globals
```
Again, not sure where this came from.

36degrees commented on packages/globals/node_modules/sass-mq/LICENSE.md:
```diff
@@ -0,0 +1,19 @@
+Copyright (c) 2013-2016 Guardian Media Group and contributors
```
This shouldn't be here.

joelanman commented:
the changelog above doesnt seem to have the govuk-button-hover-colour PR which is in the main changelog

hannalaakso commented:
Thanks @36degrees. I've got no explanation for those blank files inside `packages/all`, I've removed them. I'll note this in the docs somehow as something to look out for. 

@joelanman That would be because when I rebased master into this branch, it took your CHANGELOG message from master under `Unreleased` and moved it from  into `0.0.22-alpha (Breaking release)` as I renamed the `Unreleased` into `0.0.22-alpha (Breaking release)` in my commit. I'll mention this in the docs too. To clarify, the `govuk-button-hover-colour` PR wasn't part of this release.

alex-ju commented:
Reviewed again 👍

36degrees commented:
For posterity – there were originally a number of symlinks accidentally introduced as part of this PR (e.g. `packages/back-link/node_modules/@govuk-frontend/globals` which symlinked to `../../../globals`). These were added when the Frontend packages folder was symlinked into the Design System and `npm install` was run, and accidentally committed.

However, this was done _after_ the packages were released to npm as part of the verification process, and we have checked that the releases on npm are fine. As such, we've edited the offending commit to negate the addition of the symlinks (and a spurious .DS_Store).

We also discussed the way that commits were done – with the changes to the packages folder being committed _after_ the commit to bump the versions and release – and decided that it made more sense to commit the changes first. So there will be a further PR to update the docs along with the other changes that Hanna has been working on.
